src/value/owned.rs

```diff
@@ -63,15 +64,18 @@ impl OwnedSymbolToken {
 impl<T: Into<String>> From<T> for OwnedSymbolToken {
     /// Constructs an owned token that has only text.
     fn from(text: T) -> Self {
-        Self::new(Some(text.into()), None, None)
+        Self::new(Some(Rc::new(text.into())), None, None)
     }
 }
 
 impl SymbolToken for OwnedSymbolToken {
     type ImportSource = OwnedImportSource;
 
     fn text(&self) -> Option<&str> {
-        self.text.as_ref().map(String::as_str)
+        match self.text.as_ref() {
+            None => None,
+            Some(value) => Some(value.as_str()),
+        }
     }
```

Comment:
Personally, I have a preference for using `map` over a `match` for this because you don't have to do the un-wrapping and wrapping.  You could write it as `.map(|s| s.as_str())` for example to avoid the `match` boilerplate.